pkg/windows/windows.go

```diff
@@ -650,6 +653,10 @@ func (vm *windows) ensureServiceIsRunning(svc *service) error {
 	if err := vm.startService(svc); err != nil {
 		return errors.Wrapf(err, "error starting %s Windows service", svc.name)
 	}
+	if err := vm.setServiceDescription(svc.name); err != nil {
```

Comment:
Shouldn't this be done before the service is started? In addition:
- You should be augmenting `vm.createService()` to do this and not doing it after the fact. 
- `vm.ServiceExists()` should also be ensuring the service exists with the expected description and add it if it does not

PS: I know all this code will be removed in 6.0 but we are going to ship with this code for 5.0 and it is best it is written correctly.

Reply:
I moved the the setServiceDescription() call to before the service is started

Keeping setServiceDescription() separate from the createService func allows us to set ensure the correctness of the service description in both code paths, when the service already exists and where it needs to be created.

Additionally, it makes more sense to me to just set the service, as it would require a separate ssh command to check the value of the description. So always setting it to the proper value simplifies the code and reduces the ssh command count

Reply:
>Keeping setServiceDescription() separate from the createService func allows us to set ensure the correctness of the service description in both code paths, when the service already exists and where it needs to be created.

You can do this otherwise also. You can update `ensureServiceExists` to also check for the description and only call `setServiceDescription` only if it is not present. But I think you would need to move away from `sc.exe qe` for that which IMO is a good thing. 

>Additionally, it makes more sense to me to just set the service, as it would require a separate ssh command to check the value of the description. So always setting it to the proper value simplifies the code and reduces the ssh command count

I can make the argument that you are making an unnecessary extra SSH call all the time in the case of the service existing. Moreover you shouldn't need a separate command if you move away from `sc.exe`. Is it that complicated to use a PowerShell cmdlet instead?

Reply:
Handling this by moving away from sc.exe and editing CimInstance objects is something I considered, but I felt is out of scope. Moving away from sc.exe is deceptively challenging, as we have ensure parsing compatibility with instances using powershell as the default SSH shell, and instances using cmd. Its possible things might have changed since I did that investigation, but at that point getting powershell commands to parse properly in both cases was a huge time sink.

As a note, [Use powershell consistently throughout WMCO](https://issues.redhat.com/browse/WINC-636) has been in our backlog for a long time.

Reply:
OK, given what you are saying I will let this through but I will be a hard no on using `sc,exe` when installing WICD 😄 So please don't restate this argument in that context. 

>As a note, [Use powershell consistently throughout WMCO](https://issues.redhat.com/browse/WINC-636) has been in our backlog for a long time.

And it is going to stay that way now that we are most probably going to use the native service library in WICD, so the only place PowerShell should be used is wrt to managing WICD.

Reply:
Considering we will be using service description during the upgrade, shouldn't we retry the setting in case if set call fails?
Also, it is good to have some comments on the function description. The current description just talks about checking service and creating it if doesn't exist.

Reply:
> Considering we will be using service description during the upgrade, shouldn't we retry the setting in case if set call fails?

Retries are built into the reconciliation loop, if the description setting fails, the reconciliation will not move past that point until it succeeds.

>  Also, it is good to have some comments on the function description. The current description just talks about checking service and creating it if doesn't exist.

Changed to make the function description explicit